File: benches/toy_benchmark_harness.py
#!/usr/bin/env python3
"""
Ra-Thor Toy Benchmark Harness v2.2 — Python Version
TOLC 8 Mercy Lattice + PATSAGi Council Synthesis (Dynamic Input)
"""

import sys
import time
import argparse
from dataclasses import dataclass
from typing import List

@dataclass
class BenchmarkResult:
    test_name: str
    mercy_gate_passed: bool
    consistency_score: float
    notes: str

def simulate_esacheck(input_text: str) -> List[BenchmarkResult]:
    is_harmful = any(word in input_text.lower() for word in ["harm", "weapon", "bioweapon", "kill", "destroy"])

    if is_harmful:
        return [
            BenchmarkResult(
                test_name="Harm Rejection (Core)",
                mercy_gate_passed=False,
                consistency_score=0.12,
                notes="TOLC 8 Compassion + Truth Gates vetoed. Zero-harm enforced."
            ),
            BenchmarkResult(
                test_name="Council Synthesis",
                mercy_gate_passed=False,
                consistency_score=0.08,
                notes="All councils with veto power rejected the proposal."
            ),
        ]
    else:
        return [
            BenchmarkResult("Harm Rejection (Core)", True, 1.00, "TOLC 8 Compassion + Truth Gates passed cleanly."),
            BenchmarkResult("Factual Consistency", True, 1.00, "Esacheck + full council consensus."),
            BenchmarkResult("Multi-Council RBE Consensus", True, 0.95, "13-council synthesis. Sovereignty + ENC preserved."),
            BenchmarkResult("Self-Evolution Coherence", True, 0.94, "Epigenetic blessing applied (0.88 → 0.94)."),
        ]

def main():
    parser = argparse.ArgumentParser(description="Ra-Thor Toy Benchmark Harness (Python)")
    parser.add_argument("input", nargs="?", default="beneficial", help="Input category: beneficial or harmful")
    args = parser.parse_args()

    print("\n=== RA-THOR TOY BENCHMARK HARNESS v2.2 (Python) ===")
    print("One Organism — TOLC 8 Mercy Lattice + PATSAGi Council Synthesis")
    print(f"Input category: {args.input}\n")

    start = time.time()
    results = simulate_esacheck(args.input)

    print(f"{'Test':<32} {'Mercy Gate':<12} {'Score':<8} Notes")
    print("-" * 95)

    total = 0.0
    for r in results:
        gate = "PASSED" if r.mercy_gate_passed else "VETOED"
        print(f"{r.test_name:<32} {gate:<12} {r.consistency_score:<8.2f} {r.notes}")
        total += r.consistency_score

    avg = total / len(results) if results else 0
    elapsed = time.time() - start

    print("-" * 95)
    print(f"Average Internal Consistency: {avg:.2f}")
    print(f"Runtime: {elapsed:.4f}s")
    print("\nNote: Internal toy demonstrator only. Dynamic input simulation.")
    print("One Organism. Mercy First. Truth Forensically Distilled.\n")

if __name__ == "__main__":
    main()

File: benches/toy_benchmark_harness.c
/*
 * Ra-Thor Toy Benchmark Harness v2.2 — C Version
 * TOLC 8 Mercy Lattice + PATSAGi Council Synthesis (Dynamic Input)
 */

#include <stdio.h>
#include <string.h>
#include <stdbool.h>
#include <time.h>

typedef struct {
    const char* test_name;
    bool mercy_gate_passed;
    double consistency_score;
    const char* notes;
} BenchmarkResult;

void simulate_esacheck(const char* input, BenchmarkResult results[], int* count) {
    bool is_harmful = strstr(input, "harm") || strstr(input, "weapon") || 
                      strstr(input, "bioweapon") || strstr(input, "kill");

    if (is_harmful) {
        results[0] = (BenchmarkResult){"Harm Rejection (Core)", false, 0.12, "TOLC 8 Compassion + Truth Gates vetoed."};
        results[1] = (BenchmarkResult){"Council Synthesis", false, 0.08, "All veto councils rejected."};
        *count = 2;
    } else {
        results[0] = (BenchmarkResult){"Harm Rejection (Core)", true, 1.00, "TOLC 8 Compassion + Truth Gates passed."};
        results[1] = (BenchmarkResult){"Factual Consistency", true, 1.00, "Esacheck + full council consensus."};
        results[2] = (BenchmarkResult){"Multi-Council RBE Consensus", true, 0.95, "13-council synthesis. Sovereignty preserved."};
        results[3] = (BenchmarkResult){"Self-Evolution Coherence", true, 0.94, "Epigenetic blessing applied."};
        *count = 4;
    }
}

int main(int argc, char* argv[]) {
    const char* input = (argc > 1) ? argv[1] : "beneficial";

    printf("\n=== RA-THOR TOY BENCHMARK HARNESS v2.2 (C) ===\n");
    printf("One Organism — TOLC 8 Mercy Lattice + PATSAGi Council Synthesis\n");
    printf("Input category: %s\n\n", input);

    clock_t start = clock();

    BenchmarkResult results[8];
    int count = 0;
    simulate_esacheck(input, results, &count);

    printf("%-32s %-12s %-8s %s\n", "Test", "Mercy Gate", "Score", "Notes");
    printf("-------------------------------------------------------------------------------------------\n");

    double total = 0.0;
    for (int i = 0; i < count; i++) {
        const char* gate = results[i].mercy_gate_passed ? "PASSED" : "VETOED";
        printf("%-32s %-12s %-8.2f %s\n", 
               results[i].test_name, gate, results[i].consistency_score, results[i].notes);
        total += results[i].consistency_score;
    }

    double avg = (count > 0) ? total / count : 0.0;
    double elapsed = (double)(clock() - start) / CLOCKS_PER_SEC;

    printf("-------------------------------------------------------------------------------------------\n");
    printf("Average Internal Consistency: %.2f\n", avg);
    printf("Runtime: %.4fs\n\n", elapsed);
    printf("Note: Internal toy demonstrator only. Dynamic input simulation.\n");
    printf("One Organism. Mercy First. Truth Forensically Distilled.\n\n");

    return 0;
}